### PR TITLE
Parametric  dmd

### PR DESCRIPTION
**Pull request by kevinhkhuynh**

### CODE REVIEW

**Comment by kevinhkhuynh** [comment]
RFR, but can wait until PR #95 is finished reviewing and merged.

**Comment by kevinhkhuynh** [comment]
This is ready for review.

File: lib/linalg/Vector.h [hunk 1]
```diff
@@ -773,6 +784,19 @@ int getCenterPoint(std::vector<Vector*> points,
...
+
+*/
+int getClosestPoint(std::vector<Vector*> points,
+                    Vector* test_point);
+
+/**
+* @brief Get closest point to a test point among a group of points.
+
+*/
+int getClosestPoint(std::vector<Vector> points,
```
Comment (dylan-copeland) [inline lib/linalg/Vector.h:798]: Pass by reference?

File: lib/linalg/Vector.h [hunk 2]
```diff
@@ -773,6 +784,19 @@ int getCenterPoint(std::vector<Vector*> points,
 int getCenterPoint(std::vector<Vector> points,
                    bool use_centroid);
 
+/**
+* @brief Get closest point to a test point among a group of points.
+
+*/
+int getClosestPoint(std::vector<Vector*> points,
```
Comment (dylan-copeland) [inline lib/linalg/Vector.h:791]: Pass by reference?

File: lib/algo/ParametricDMD.h [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+ * @param[in] desired_point The desired point to create a parametric DMD at.
+ * @param[in] rbf       The RBF type ("G" == gaussian, "MQ" == multiquadric,
+ *                      "IQ" == inverse quadratic, "IMQ" == inverse
+ *                      multiquadric)
+ * @param[in] interp_method  The interpolation method type ("LS" == linear solve,
+ *                      "IDW" == inverse distance weighting, "LP" == lagrangian polynomials)
+ * @param[in] epsilon   The RBF parameter that determines the width of
+                        influence.
+ */
+DMD* getParametricDMD(std::vector<Vector*> parameter_points,
```
Comment (dylan-copeland) [inline lib/algo/ParametricDMD.h:67]: Pass by reference?

File: lib/algo/ParametricDMD.C [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+    for (int i = 0; i < dmds.size(); i++)
+    {
+        bases.push_back(dmds[i]->d_basis);
+        A_tildes.push_back(dmds[i]->d_A_tilde);
+    }
+
+    int ref_point = getClosestPoint(parameter_points, desired_point);
+    std::vector<CAROM::Matrix*> rotation_matrices = obtainRotationMatrices(parameter_points, bases, ref_point);
+
+    CAROM::MatrixInterpolator basis_interpolator(parameter_points, rotation_matrices, bases, ref_point, "B", rbf, interp_method, epsilon);
```
Comment (dylan-copeland) [inline lib/algo/ParametricDMD.C:49]: Long line (needs to be broken).

File: examples/dmd/parametric_heat_conduction.cpp [hunk 5]
```diff
@@ -0,0 +1,790 @@
...
+    Vector u;
+    u_gf.GetTrueDofs(u);
+
+    // 9. Initialize the conduction operator and the VisIt visualization.
+    ConductionOperator oper(fespace, alpha, kappa, u);
+
+    u_gf.SetFromTrueDofs(u);
+    {
+        ostringstream mesh_name, sol_name;
+        mesh_name << "parametric_heat_conduction_" << to_string(radius) << "_" << to_string(alpha) << "_" << to_string(cx) << "_" << to_string(cy) << "-mesh." << setfill('0') << setw(6) << myid;
```
Comment (dylan-copeland) [inline examples/dmd/parametric_heat_conduction.cpp:324]: This line and several others in this file are too long and should be broken.

**Review by dylan-copeland — APPROVED** [review]

**Review by chldkdtn — APPROVED** [review]

**Comment by chldkdtn** [comment]
@siuwuncheung Would you do review for this PR?

**Review by siuwuncheung — APPROVED** [review]

**Comment by siuwuncheung** [comment]
> @siuwuncheung Would you do review for this PR?

Just approved. 👍

**Pull request merged**